lms/djangoapps/courseware/views/views.py

```diff
@@ -1014,7 +1014,7 @@ def dates(request, course_id):
     from lms.urls import COURSE_DATES_NAME, RESET_COURSE_DEADLINES_NAME
 
     course_key = CourseKey.from_string(course_id)
-    if course_home_mfe_dates_tab_is_active(course_key) and not request.user.is_staff:
+    if course_home_mfe_dates_tab_is_active(course_key) and not request.user.is_staff and not course_key.deprecated:
```

Comment:
Should we move that check into `course_home_mfe_dates_tab_is_active`?

Reply:
On that note, might be even better to just add into the condition for course_home_mfe in general (the bigger flag and not the per-tab)... Or else implemented on each of the tabs

Reply:
I was wondering this too... the file looked so clean I wasn't sure if I should drag non-waffle flag logic into it. But I'd say yes?